src/IBM.Cloud.SDK.Core/Util/CredentialUtils.cs

```diff
@@ -349,6 +349,17 @@ private static void AddToDictionary(Dictionary<string, string> dictionary, strin
                 string key = lineTokens[0];
                 string value = lineTokens[1];
 
+                // serviceName and the credentials key must use the same casing.
+                // These two if statements ensure that the serviceName passed matches the credential's key casing
+                if(key.Contains("-") && serviceName.Contains("_"))
+                {
+                    serviceName = serviceName.Replace('_', '-');
+                }
+                else if (key.StartsWith(serviceName.Replace('-','_')) && serviceName.Contains("-"))

```

Comment:
why are we using startsWith here, but in the previous if it uses contains

Reply:
This is because the "key" will always contain a '_' (ex. NATURAL-LANGUAGE-UNDERSTANDING_APIKEY) so I must be more specific and check that only the service name itself uses '_'